rmw_connext_shared_cpp/src/topic_endpoint_info.cpp

```diff
@@ -12,32 +12,313 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include <string>
 #include <map>
+#include <vector>
 
+#include "rmw/error_handling.h"
+#include "rmw/impl/cpp/key_value.hpp"
+
+#include "rmw_connext_shared_cpp/demangle.hpp"
+#include "rmw_connext_shared_cpp/namespace_prefix.hpp"
 #include "rmw_connext_shared_cpp/topic_endpoint_info.hpp"
+#include "rmw_connext_shared_cpp/types.hpp"
 
-#include "rmw/error_handling.h"
+struct ParticipantNameInfo
+{
+  std::string name;
+  std::string namespace_;
+};
+
+rmw_ret_t
+_validate_params(
+  const char * identifier,
+  const rmw_node_t * node,
+  rcutils_allocator_t * allocator,
+  const char * topic_name,
+  rmw_topic_endpoint_info_array_t * participants_info)
+{
+  if (nullptr == identifier) {
```

Comment:
nit: You can use https://github.com/ros2/rmw/blob/28f91855510c3c832b500fae40fdcc4169d26a0d/rmw/include/rmw/error_handling.h#L35.

Reply:
Done.